lib/graphql/static_validation/rules/variable_usages_are_allowed.rb

```diff
@@ -66,6 +71,29 @@ def create_error(error_message, var_type, ast_var, arg_defn, arg_node, context)
         message("#{error_message} on variable $#{ast_var.name} and argument #{arg_node.name} (#{var_type.to_s} / #{arg_defn.type.to_s})", arg_node, context: context)
       end
 
+      def wrap_var_type_with_depth_of_arg(var_type, arg_node)
+        arg_node_value = arg_node.value
+        return var_type unless arg_node_value.is_a?(Array)
+        new_var_type = var_type
+
+        array_depth = 1 + depth_of_array(arg_node_value)
+
+        array_depth.times do
+          new_var_type = GraphQL::ListType.new(of_type: new_var_type)
+        end
+
+        new_var_type
+      end
+
+      def depth_of_array(some_array)
+        result = 0
+        while (flattened = some_array.flatten) != some_array
+          result += 1
+          some_array = flattened
+        end
+        result
+      end
```

Comment:
I've used your suggestion. Alternatively, could go for a solution that uses `flatten!` instead. That avoids the recursive method call, but would need to run benchmarks to see if that's helpful or not.